components/server/src/server.rs

```diff
@@ -1841,23 +1844,29 @@ impl<EK: KvEngine, R: RaftEngine> EngineMetricsManager<EK, R> {
 }
 
 pub struct EnginesResourceInfo {
-    kv_engine: RocksEngine,
+    tablet_factory: Arc<KvEngineFactory>,
     raft_engine: Option<RocksEngine>,
+    // region_id -> (suffix, tablet)
+    // `update` is called perodically which needs this map for recording the latest tablet for each
+    // region and cached_latest_tablets is used to avoid memory allocation each time when
+    // calling `update`.
+    cached_latest_tablets: Arc<Mutex<HashMap<u64, (u64, RocksEngine)>>>,
```

Comment:
Why not make `update` accept a map? So lock is not required.

Reply:
So who passes the map to `update` ? I think acquiring lock here is very fast and no contention will be involved.

Reply:
```
let engines_info_clone = engines_info.clone();
let mut map = HashMap::default();
self.background_worker
    .spawn_interval_task(DEFAULT_METRICS_FLUSH_INTERVAL, move || {
        let now = Instant::now();
        engine_metrics.flush(now);
        io_metrics.flush(now);
        engines_info_clone.update(now, &mut map);
    });
```

If lock is not used for solving conflicts, it means the implementation is wrong. Think twice before giving up.

Reply:
Okay, good suggestion. I will propose another PR for it.